sklearn/compose/_column_transformer.py

```diff
@@ -473,6 +522,68 @@ def transform(self, X):
 
         return _hstack(list(Xs), self.sparse_output_)
 
+    def inverse_transform(self, X):
+        """Apply inverse transform
+
+        All estimators in ``transformers`` must support ``inverse_transform``.
+
+        Parameters
+        ----------
+        X: array-like, shape = [n_samples, n_transformed_features]
+            Data samples, where ``n_samples`` is the number of samples and
+            ``n_features`` is the number of features.
+
+        Returns
+        -------
+        Xt : array-like or DataFrame of shape [n_samples, n_features]
+
+        """
+        check_is_fitted(self, 'transformers_')
+
+        if not self._invertible[0]:
+            raise ValueError("Unable to invert: ".format(
+                self._invertible[1]))
+
+        X_subs = (_get_column(X, key) for key in self._output_indices)
+        inv_transformers = []
+        get_weight = (self.transformer_weights or {}).get
+
+        for (name, trans, column), sub in zip(self.transformers_, X_subs):
+            if trans == 'passthrough':
+                trans = FunctionTransformer(
+                    validate=False, accept_sparse=True, check_inverse=False)
+
+            inv_transformers.append((name, trans, sub, get_weight(name)))
+
+        Xs = Parallel(n_jobs=self.n_jobs)(
+            delayed(_inverse_transform_one)(
+                trans, X_sel, weight)
+            for _, trans, X_sel, weight in inv_transformers)
+
+        if not Xs:
+            # All transformers are None
+            return np.zeros((X.shape[0], 0))
+
+        inverse_Xs = np.zeros((Xs[0].shape[0], self._X_features))
+        for indices, inverse_X in zip(self._input_indices, Xs):
+            if sparse.issparse(inverse_X):
+                inverse_Xs[:, indices] = inverse_X.toarray()
+            else:
+                if inverse_X.ndim == 1:
+                    inverse_Xs[:, indices] = inverse_X[:, np.newaxis]
+                else:
+                    inverse_Xs[:, indices] = inverse_X
+
+        if self._X_is_sparse:
+            return sparse.csr_matrix(inverse_Xs)
+        if self._X_columns is not None:
+            try:
+                import pandas as pd
+                return pd.DataFrame(inverse_Xs, columns=self._X_columns)
+            except ImportError:
```

Comment:
This makes no sense. Either we promise pandas or we don't. Changing the return type on the basis of not having a dependency can't happen.